editor/store/effects/posts.js

```diff
@@ -0,0 +1,315 @@
+	if ( isAutosave ) {
+		return;
+	}
+
+	const publishStatus = [ 'publish', 'private', 'future' ];
+	const isPublished = includes( publishStatus, previousPost.status );
+	const willPublish = includes( publishStatus, post.status );
+
+	let noticeMessage;
+	let shouldShowLink = true;
+
+	if ( ! isPublished && ! willPublish ) {
+		// If saving a non-published post, don't show notice.
+		noticeMessage = null;
+	} else if ( isPublished && ! willPublish ) {
+		// If undoing publish status, show specific notice
+		noticeMessage = __( 'Post reverted to draft.' );
+		shouldShowLink = false;
+	} else if ( ! isPublished && willPublish ) {
+		// If publishing or scheduling a post, show the corresponding
+		// publish message
+		noticeMessage = {
+			publish: __( 'Post published!' ),
+			private: __( 'Post published privately!' ),
+			future: __( 'Post scheduled!' ),
+		}[ post.status ];
+	} else {
+		// Generic fallback notice
+		noticeMessage = __( 'Post updated!' );
+	}
+
+	if ( noticeMessage ) {
+		dispatch( createSuccessNotice(
+			<p>
+				{ noticeMessage }
+				{ ' ' }
+				{ shouldShowLink && <a href={ post.link }>{ __( 'View post' ) }</a> }
+			</p>,
+			{ id: SAVE_POST_NOTICE_ID, spokenMessage: noticeMessage }
+		) );
+	}
+};
+
+/**
+ * Request Post Update Failure Effect handler
+ *
+ * @param {Object} action  action object.
+ * @param {Object} store   Redux Store.
+ */
+export const requestPostUpdateFailure = ( action, store ) => {
+	const { post, edits, error } = action;
+
+	if ( error && 'rest_autosave_no_changes' === error.code ) {
+		// Autosave requested a new autosave, but there were no changes. This shouldn't
+		// result in an error notice for the user.
+		return;
+	}
+
+	const { dispatch } = store;
+
+	const publishStatus = [ 'publish', 'private', 'future' ];
+	const isPublished = publishStatus.indexOf( post.status ) !== -1;
+	// If the post was being published, we show the corresponding publish error message
+	// Unless we publish an "updating failed" message
+	const messages = {
+		publish: __( 'Publishing failed' ),
+		private: __( 'Publishing failed' ),
+		future: __( 'Scheduling failed' ),
+	};
+	const noticeMessage = ! isPublished && publishStatus.indexOf( edits.status ) !== -1 ?
+		messages[ edits.status ] :
+		__( 'Updating failed' );
+	dispatch( createErrorNotice( noticeMessage, { id: SAVE_POST_NOTICE_ID } ) );
+};
+
+/**
+ * Trash Post Effect handler
+ *
+ * @param {Object} action  action object.
+ * @param {Object} store   Redux Store.
+ */
+export const trashPost = async ( action, store ) => {
+	const { dispatch, getState } = store;
+	const { postId } = action;
+	const postTypeSlug = getCurrentPostType( getState() );
+	const postType = await resolveSelector( 'core', 'getPostType', postTypeSlug );
+
+	dispatch( removeNotice( TRASH_POST_NOTICE_ID ) );
+	try {
+		await apiFetch( { path: `/wp/v2/${ postType.rest_base }/${ postId }`, method: 'DELETE' } );
+	} catch ( error ) {
```

Comment:
I see this code missing after `apiFetch` when comparing with master:
```js
const post = getCurrentPost( getState() );

// TODO: This should be an updatePost action (updating subsets of post properties),
// But right now editPost is tied with change detection.
dispatch( resetPost( { ...post, status: 'trash' } ) );
```

Is that expected change?

Reply:
No it's not, I wonder how this happened :P